store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedUtils.java

```diff
@@ -0,0 +1,153 @@
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class with static methods to help support the federated store.
+ */
+public final class FederatedUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FederatedUtils.class);
+
+    private FederatedUtils() {
+        // utility class
+    }
+
+    /**
+     * Checks if graphs share any groups between their schemas.
+     *
+     * @param graphs Graphs to check.
+     * @return Do they share groups.
+     */
+    public static boolean doGraphsShareGroups(final List<GraphSerialisable> graphs) {
+        // Check if any of the graphs have common groups in their schemas
+        List<Schema> schemas = graphs.stream()
+            .map(GraphSerialisable::getSchema)
+            .collect(Collectors.toList());
+
+        // Compare all schemas against each other
+        for (int i = 0; i < schemas.size() - 1; i++) {
+            for (int j = i + 1; j < schemas.size(); j++) {
+                // Compare each schema against the others to see if common groups
+                if (!Collections.disjoint(schemas.get(i).getGroups(), schemas.get(j).getGroups())) {
+                    LOGGER.debug("Found common schema groups between requested graphs");
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Get a version of the operation chain that satisfies the schema for the
+     * requested graph. This will ensure the operation will pass validation on
+     * the sub graph end as there is no universal way to skip validation.
+     *
+     * @param operation The operation.
+     * @param graphSerialisable The graph.
+     * @param depth Current recursion depth of this method.
+     * @return A valid version of the operation chain.
+     */
+    public static OperationChain getValidOperationForGraph(final Operation operation, final GraphSerialisable graphSerialisable, final int depth) {
+        LOGGER.debug("Creating valid operation for graph, depth is: {}", depth);
+        final Collection<Operation> updatedOperations = new ArrayList<>();
+
+        // Fix the view so it will pass schema validation
+        if (operation instanceof OperationView
+                && ((OperationView) operation).getView() != null
+                && ((OperationView) operation).getView().hasGroups()) {
+
+            // Update the view for the graph
+            ((OperationView) operation).setView(
+                getValidViewForGraph(((OperationView) operation).getView(), graphSerialisable));
+
+            updatedOperations.add(operation);
+
+        // Recursively go into operation chains to make sure everything is fixed
+        } else if (operation instanceof OperationChain) {
+            for (final Operation op : ((OperationChain<?>) operation).getOperations()) {
+                // Resolve if haven't hit the depth limit for validation
+                if (depth < 5) {
```

Comment:
Shall we make this magic number a constant at least? or maybe configurable

Reply:
Good point could make it an operation option with a default